azurerm/internal/services/appplatform/tests/resource_arm_spring_cloud_config_server_test.go

```diff
@@ -0,0 +1,225 @@
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.search_paths.#", "0"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.#", "0"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMSpringCloudConfigServer_public(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_spring_cloud_config_server", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMSpringCloudDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMSpringCloudConfigServer_public(data),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(data.ResourceName, "id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.uri", "https://github.com/Azure-Samples/piggymetrics"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.label", "config"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.search_paths.#", "2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.search_paths.0", "dir1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.search_paths.1", "dir2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.#", "2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.name", "repo1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.uri", "https://github.com/Azure-Samples/piggymetrics"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.label", "config"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.search_paths.#", "2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.search_paths.0", "dir1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.0.search_paths.1", "dir2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.name", "repo2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.uri", "https://github.com/Azure-Samples/piggymetrics"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.label", "config"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.search_paths.#", "2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.search_paths.0", "dir1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "git_settings.0.repositories.1.search_paths.1", "dir2"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMSpringCloudConfigServer_httpBasicAuth(t *testing.T) {
```

Comment:
this case has been run. `username` and `password` are both returned "*".
Because in previous comments, the schema fields should have its context.
I wraped these two fields in "http_basic_auth" nested block, and wrap another four fields in "ssh_auth" nested block